### PR TITLE
feat: add basic form example

### PR DESCRIPTION
**Pull request by KyleAMathews**

### CODE REVIEW

File: examples/basic-form/src/pages/index.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        style={{ display: `block`, marginBottom: 16 }}
+        {...register("Email", { required: true, pattern: /^\S+@\S+$/i })}
+      />
+
+      <label htmlFor="tel">Mobile number</label>
+      <input
+        id="tel"
+        type="tel"
+        style={{ display: `block`, marginBottom: 16 }}
+        {...register("Mobile number", {
```
Comment (DSchau) [inline examples/basic-form/src/pages/index.js:60]: @KyleAMathews this is a lot better than my janky code 😆

Comment (KyleAMathews) [reply examples/basic-form/src/pages/index.js:60]: haha yeah, that's the goal — give people something nice to copy/paste

File: examples/basic-form/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Gatsby basic form Functions example
+
+This example shows how to build a form with [react-hook-form](https://react-hook-form.com/) that submits to a Gatsby Function.
+
+1.  **Start developing.**
+
+    To get started clone this repo locally and run `yarn` to add all necessary packages.
+
+    ```shell
+    cd hello-world
```
Comment (dmccraw) [inline examples/basic-form/README.md:10]: cd basic-form?

Comment (KyleAMathews) [reply examples/basic-form/README.md]: yes, fixed

**Review by dmccraw — APPROVED** [review]
:ship: it!

**Pull request merged**